### DISCUSSION TITLE
QMK support for crkbd rev4

### DISCUSSION THREAD

**Discussion by jarviliam** [manna-harbour/miryoku#311] [Q&A] [answered]
I've recently changed my keyboard to the new crkbd revision and I've run into some issues trying to compile miryoku for it as I'm not too knowledgeable about qmk.

Here is the new crkbd firmware for reference:
https://github.com/foostan/kbd_firmware/tree/main/keyboards/crkbd/qmk/qmk_firmware/rev4

I attempted to basically update the crkbd keyboard folder with the rev4 one from the link above and compile it. I ran into issues with `matrix_size` not being defined but more importantly it seems the info.json that the rev4 uses has some new features not compatible with the qmk in this repo. Namely `transport.sync` (https://github.com/foostan/kbd_firmware/blob/a408165b0898ee825cb48925e40de1d8ae208c59/keyboards/crkbd/qmk/qmk_firmware/rev4/info.json#L67) being unknown. This results in the info.json file being ignored, which I believe is causing the matrix_size error?

Any idea on how I can get this working with rev4?

**Comment by jarviliam** [comment, marked as answer]
Actually, I managed to solve it after diving into qmk a bit more. For anyone else trying to do the same, here's what I did. 

1) Checkout the qmk_firmware from the last commit before they removed user keymaps. https://github.com/manna-harbour/miryoku/discussions/287

2) Remove the old crkbd folder from keyboards/crkbd and copy over the rev4 firmware into keyboards/crkbd

3) Edit the info.json of the rev4 and change the sync field under transport to sync_matrix_state with a value of true.

That should allow you to compile it

**Reply by manna-harbour** [reply]
Alternatively, try the link for the "qmk_userspace" version at https://github.com/manna-harbour/miryoku/discussions/287.
